app/org/apache/spark/deploy/history/SparkFSFetcher.scala

```diff
@@ -50,11 +52,19 @@ class SparkFSFetcher(fetcherConfData: FetcherConfigurationData) extends Elephant
   if (fetcherConfData.getParamMap.get(LOG_SIZE_XML_FIELD) != null) {
     val logLimitSize = Utils.getParam(fetcherConfData.getParamMap.get(LOG_SIZE_XML_FIELD), 1)
     if (logLimitSize != null) {
-      EVENT_LOG_SIZE_LIMIT_MB = MemoryFormatUtils.stringToBytes(logLimitSize(0) + "M");
+      EVENT_LOG_SIZE_LIMIT_MB = logLimitSize(0)
     }
   }
   logger.info("The event log limit of Spark application is set to " + EVENT_LOG_SIZE_LIMIT_MB + " MB")
 
+  if (fetcherConfData.getParamMap.get(LOG_DIR_XML_FIELD) != null) {
+    val logDir = fetcherConfData.getParamMap.get(LOG_DIR_XML_FIELD)
+    if (!logDir.isEmpty) {
+      DEFAULT_LOG_DIR = logDir
```

Comment:
If it's default, it shouldn't be changed. Can you use a different variable name ? Also, all caps normally indicates a constant to me.